src/repositories/books/books.repository.ts

```diff
@@ -1,19 +1,52 @@
-import { readFile } from '../../helpers/helpers.ts';
-import { Book } from '../../common/types/types.ts';
+import { v4 } from 'https://deno.land/std@0.98.0/uuid/mod.ts';
+import { readFile, writeFile } from '../../helpers/helpers.ts';
+import { Book } from '../../common/models/models.ts';
 import { IRepository } from '../../common/interfaces/interfaces.ts';
 
-const booksDataPath = new URL('./books.json', import.meta.url).pathname;
-
-class Books implements Partial<IRepository<Book>> {
+const booksDataPath = new URL('./books.json', import.meta.url);
+export class Books implements IRepository<Book> {
   public findAll(): Promise<Book[]> {
     return this._getBooks();
   }
 
   private async _getBooks(): Promise<Book[]> {
     const books = await readFile(booksDataPath);
-
     return JSON.parse(books);
   }
+
+  public async findOne(id: string): Promise<Book | null> {
+    const books = await this._getBooks();
+    return books.find(book => book.id === id) || null;
```

Comment:
It would be better to convert all these things into function and move them to `src/repositories/books/helers/`